e2e/next-sandbox/test/utils.ts

```diff
@@ -76,6 +84,51 @@ export function delay(ms: number) {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
+type TimeoutOptions = {
+  timeout?: number;
+  step?: number;
+};
+
+const DEFAULT_TIMEOUT = 1500;
+const DEFAULT_STEP = 60;
+
+export async function waitForTextContent(
+  page: Page,
+  id: string,
+  expectedText: string,
+  options?: TimeoutOptions
+) {
+  const start = Date.now();
+  const timeoutAt = start + (options?.timeout ?? DEFAULT_TIMEOUT);
+  const attempts = [];
+
+  do {
+    const foundText = await getTextContentOrEmpty(page, id);
+    const ms = Date.now() - start;
+    if (!foundText) {
+      attempts.push(`(after ${ms}ms) Element with id ${id} not found yet`);
+    } else if (foundText !== expectedText) {
+      attempts.push(
+        `(after ${ms}ms) Element did not contain expected text yet: ${JSON.stringify(
+          foundText
+        )}`
+      );
+    } else {
+      // Done!
+      return;
+    }
+    await delay(options?.step ?? DEFAULT_STEP);
+  } while (Date.now() < timeoutAt);
+
+  const ms = Date.now() - start;
+  attempts.push(`(after ${ms}ms) Timed out`);
+  throw new Error(
+    `Expected text content was never found\n\nid: ${id}\nI tried looking for: ${JSON.stringify(
+      expectedText
+    )}\n\nHere were my attempts:\n${attempts.join("\n")}`
+  );
+}
```

Comment:
The tl;dr here is: try to find a text in a loop. Polling every 60ms, until a timeout of 1500ms is exceeded. Stop as soon as the text is found. If never found, show a log tail of what was attempted and what was the found text (this _hugely_ helps debugging/writing the tests).